src/components/imageUploadAgent.js

```diff
@@ -0,0 +1,93 @@
+import * as ImagePicker from "expo-image-picker";
+import React, { useState, useEffect } from "react";
+import {
+  View,
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+  Platform,
+  Text,
+} from "react-native";
+
+import { COLOURS } from "../constants";
+const ImageUploadAgent = ({ onImageUpload }) => {
+  const [image, setImage] = useState(null);
+
+  useEffect(() => {
+    // Request permission to access the device's gallery
+    (async () => {
+      if (Platform.OS !== "web") {
+        const { status } =
+          await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== "granted") {
+          alert("Sorry, we need camera roll permissions to make this work!");
```

Comment:
This message should be changed, something like to proceed grant permission to access your photo gallery.

Reply:
Done